src/app/utilities/signals-manager.utility.ts

```diff
@@ -21,7 +21,7 @@ export class SignalsManager<T extends { [K in keyof T]: any }> {
     this.signalsCollection.set(key, signalObject);
   }
 
-  getSignal(key: keyof T): WritableSignal<T[keyof T]> {
+  getSignal<U>(key: keyof T): WritableSignal<U> {
```

Comment:
One change I recommend here, is to change this U generic type into something dependant of the T one, that way we manage at least that is has some sort of relationship between them

```Typescript getSignal<U extends T[keyof T]>(key: keyof T): WritableSignal<U> {```